openhcl/virt_mshv_vtl/src/processor/tdx/mod.rs

```diff
@@ -3445,22 +3445,19 @@ impl<T: CpuIo> hv1_hypercall::FlushVirtualAddressListEx
         {
             let mut flush_state = self.vp.shared.flush_state[vtl].write();
 
-            // If there are too many provided gvas then promote this request to a flush entire.
-            // TODO TDX GUEST VSM do we need the extended check? I don't think so
-            if gva_ranges.len() > FLUSH_GVA_LIST_SIZE {
+            // If we fail to add ranges to the list for any reason then promote this request to a flush entire.
+            if Self::add_ranges_to_tlb_flush_list(
+                &mut flush_state,
+                gva_ranges,
+                flags.use_extended_range_format(),
+            )
+            .is_err()
```

Comment:
nit: I would write `if let Err(()) = ...` so that you get a compiler error if you add an error type in the future, so that you don't forget to update the caller(s) to handle the error payload appropriately.